modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -67,7 +67,7 @@ private long parseMillis(String date) {
         @Override
         Function<String, DateTime> getFunction(String format, DateTimeZone timezone, Locale locale) {
             DateTimeFormatter parser = DateTimeFormat.forPattern(format).withZone(timezone).withLocale(locale);
-            return text -> parser.withDefaultYear((new DateTime(DateTimeZone.UTC)).getYear()).parseDateTime(text);
+            return text -> parser.withDefaultYear((new DateTime(DateTimeZone.UTC)).getYear()).parseDateTime(text.toLowerCase());
```

Comment:
This is an attempt to make the date parsing case insensitive. If you remove this change then you will be able to parse "mai 2017" but not "Mai 2017" for example.